### PR TITLE
Adding details to Conformance Tests using RFC 2119 standards.

### PR DESCRIPTION
**Pull request by brahmaroutu**
This PR is part of the conformance documentation. This is to provide more formal specification using RFC 2119 keywords to describe the test so that who ever is running conformance tests do not have to go through the code to understand why and what is tested.
The documentation information added here into each of the tests eventually result into a document which is currently checked in at location https://github.com/cncf/k8s-conformance/blob/master/docs/KubeConformance-1.9.md

I would like to have this PR reviewed for v1.10 as I consider it important to strengthen the conformance documents.

### CODE REVIEW

**Comment by BenTheElder** [comment]
flake is https://github.com/kubernetes/kubernetes/issues/60614

**Comment by brahmaroutu** [comment]
@WilliamDenniss @duglin @bradtopol Please review conformance documentation.

File: test/e2e/auth/service_accounts.go [hunk 1]
```diff
@@ -153,6 +153,11 @@ var _ = SIGDescribe("ServiceAccounts", func() {
 		}
 	})
 
+	/*
+		Release : v1.9
```
Comment (tallclair) [inline test/e2e/auth/service_accounts.go]: nit: extra space before `:`

File: test/e2e/auth/service_accounts.go [hunk 2]
```diff
@@ -153,6 +153,11 @@ var _ = SIGDescribe("ServiceAccounts", func() {
...
+		Testname: Service Account Tokens Must AutoMount
+		Description: Ensure that Service Account keys are  mounted into the Pod. Pod contains three containers each will read Service Account Key, root CA Key and Namespace Key respectively from the default API Token Mount path. All these three files MUST exist and the Service Account mount path MUST be auto mounted to the Pod.
```
Comment (tallclair) [inline test/e2e/auth/service_accounts.go]: Please wrap this line around 80-100 characters

File: test/e2e/auth/service_accounts.go [hunk 3]
```diff
@@ -235,6 +240,25 @@ var _ = SIGDescribe("ServiceAccounts", func() {
 		})
 	})
 
+	/*
+		Release : v1.9
+		Testname: Service account tokens auto mount optionally
+		Description: Ensure that Service Account keys are  mounted into the Pod only when AutoMountServiceToken is not set to false. We test the following scenarios here.
```
Comment (tallclair) [inline test/e2e/auth/service_accounts.go]: ditto: please wrap

**Comment by brahmaroutu** [comment]
@tallclair Please review.

File: test/e2e/auth/service_accounts.go [hunk 4]
```diff
@@ -153,6 +153,14 @@ var _ = SIGDescribe("ServiceAccounts", func() {
...
 	})
 
+	/*
+				Release: v1.9
+				Testname: Service Account Tokens Must AutoMount
+				Description: Ensure that Service Account keys are  mounted into the Pod. Pod contains
+		 three containers each will read Service Account Key, root CA Key and Namespace Key respectively from
+		 the default API Token Mount path. All these three files MUST exist and the Service Account mount
+		path MUST be auto mounted to the Pod.
+	*/
```
Comment (tallclair) [inline test/e2e/auth/service_accounts.go]: Please fix the indentation.

File: test/e2e/auth/service_accounts.go [hunk 5]
```diff
@@ -153,6 +153,14 @@ var _ = SIGDescribe("ServiceAccounts", func() {
 		}
 	})
 
+	/*
+				Release: v1.9
+				Testname: Service Account Tokens Must AutoMount
```
Comment (tallclair) [inline test/e2e/auth/service_accounts.go]: Why is the documented name different from the code description, i.e. "ServiceAccounts should mount an API token into pods". Seems like they should be consistent...

File: test/e2e/auth/service_accounts.go [hunk 6]
```diff
@@ -153,6 +153,14 @@ var _ = SIGDescribe("ServiceAccounts", func() {
 		}
 	})
 
+	/*
+				Release: v1.9
```
Comment (tallclair) [inline test/e2e/auth/service_accounts.go]: Why 1.9? What's the significance of this field?

File: test/e2e/auth/service_accounts.go [hunk 7]
```diff
@@ -153,6 +153,14 @@ var _ = SIGDescribe("ServiceAccounts", func() {
...
+				Testname: Service Account Tokens Must AutoMount
+				Description: Ensure that Service Account keys are  mounted into the Pod. Pod contains
+		 three containers each will read Service Account Key, root CA Key and Namespace Key respectively from
```
Comment (tallclair) [inline test/e2e/auth/service_accounts.go]: remove `Key` - that's just an artifact of code reuse: `... containers will each read the Service Account token, root CA, and  default namespace`

File: test/e2e/auth/service_accounts.go [hunk 8]
```diff
@@ -153,6 +153,14 @@ var _ = SIGDescribe("ServiceAccounts", func() {
...
+		 the default API Token Mount path. All these three files MUST exist and the Service Account mount
+		path MUST be auto mounted to the Pod.
```
Comment (tallclair) [inline test/e2e/auth/service_accounts.go]: nit: s/Pod/Container/

File: test/e2e/auth/service_accounts.go [hunk 9]
```diff
@@ -235,7 +243,31 @@ var _ = SIGDescribe("ServiceAccounts", func() {
 		})
 	})
 
+	/*
+				Release: v1.9
+				Testname: Service account tokens auto mount optionally
+				Description: Ensure that Service Account keys are  mounted into the Pod only
+		 when AutoMountServiceToken is not set to false. We test the following scenarios here.
+				1. Create Pod,  Pod Spec has  AutomountServiceAccountToken  set to nil
```
Comment (tallclair) [inline test/e2e/auth/service_accounts.go]: Fix spacing, same below.

File: test/e2e/auth/service_accounts.go [hunk 10]
```diff
@@ -235,7 +243,31 @@ var _ = SIGDescribe("ServiceAccounts", func() {
...
+					a) Service Account is default,
```
Comment (tallclair) [inline test/e2e/auth/service_accounts.go]: What does this mean?

File: test/e2e/auth/service_accounts.go [hunk 11]
```diff
@@ -235,7 +243,31 @@ var _ = SIGDescribe("ServiceAccounts", func() {
...
+					b) Service Account is an object with AutomountServiceAccountToken set to true,
```
Comment (tallclair) [inline test/e2e/auth/service_accounts.go]: What does "Service Account is an object" mean?

File: test/e2e/auth/service_accounts.go (hunk 5 above)
Comment (brahmaroutu) [reply test/e2e/auth/service_accounts.go]: @tallclair  I was just trying to generalize all 'service account token' related tests with that prefix. Please suggest better wording.
Service Account Tokens - must auto mount
Service Account Tokens - auto mount optionally ...

File: test/e2e/auth/service_accounts.go (hunk 6 above)
Comment (brahmaroutu) [reply test/e2e/auth/service_accounts.go]: This field will be used to track the conformance tests by release. First introduced in 1.9 and then updates the test in 1.12, etc suggesting that this test could potentially impact a conformance run in 1.12 release.

File: test/e2e/auth/service_accounts.go (hunk 10 above)
Comment (brahmaroutu) [reply test/e2e/auth/service_accounts.go]: I meant no changes to out-of-the box behavior without configuration.

File: test/e2e/auth/service_accounts.go (hunk 6 above)
Comment (tallclair) [reply test/e2e/auth/service_accounts.go]: So this is the release the conformance test was introduced in? (As opposed to when the feature was introduced?)

**Comment by brahmaroutu** [comment]
@tallclair Can you please review.

File: test/e2e/auth/service_accounts.go (hunk 4 above)
Comment (tallclair) [reply test/e2e/auth/service_accounts.go]: Still not fixed

File: test/e2e/auth/service_accounts.go (hunk 5 above)
Comment (tallclair) [reply test/e2e/auth/service_accounts.go]: Lots conformance tests replace spaces with dashes in the testname. Is the expected value / format / purpose of the testname documented anywhere?

File: test/e2e/auth/service_accounts.go (hunk 4 above)
Comment (brahmaroutu) [reply test/e2e/auth/service_accounts.go]: Sorry about that.

**Comment by brahmaroutu** [comment]
@tallclair can you please review.

File: test/e2e/auth/service_accounts.go (hunk 4 above)
Comment (tallclair) [reply test/e2e/auth/service_accounts.go]: Still not fixed.

File: test/e2e/auth/service_accounts.go (hunk 6 above)
Comment (brahmaroutu) [reply test/e2e/auth/service_accounts.go]: yes, intention is to track conformance.

File: test/e2e/auth/service_accounts.go (hunk 5 above)
Comment (brahmaroutu) [reply test/e2e/auth/service_accounts.go]: This test name here is the header in the conformance document for the test and format is a readable textual string.

**Comment by tallclair** [comment]
You have still not addressed the comments from my last review. Please address all comments before requesting another review.

File: test/e2e/auth/service_accounts.go (hunk 1 above)
Comment (brahmaroutu) [reply test/e2e/auth/service_accounts.go]: fixed

File: test/e2e/auth/service_accounts.go (hunk 2 above)
Comment (brahmaroutu) [reply test/e2e/auth/service_accounts.go]: I think gofmt was messing up with some of the indentation, it seem to be smart with comment body too. Please see if I addressed all your concerns.

File: test/e2e/auth/service_accounts.go (hunk 4 above)
Comment (brahmaroutu) [reply test/e2e/auth/service_accounts.go]: fixed.

File: test/e2e/auth/service_accounts.go (hunk 7 above)
Comment (brahmaroutu) [reply test/e2e/auth/service_accounts.go]: fixed.

File: test/e2e/auth/service_accounts.go (hunk 9 above)
Comment (brahmaroutu) [reply test/e2e/auth/service_accounts.go]: fixed

File: test/e2e/auth/service_accounts.go (hunk 8 above)
Comment (brahmaroutu) [reply test/e2e/auth/service_accounts.go]: fixed.

**Pull request merged**